src/ShipmentListBuilder.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Drupal\commerce_shipping;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\EntityListBuilder;
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Defines the list builder for shipments.
+ */
+class ShipmentListBuilder extends EntityListBuilder {
+
+  /**
+   * The current route match.
+   *
+   * @var \Drupal\Core\Routing\RouteMatchInterface
+   */
+  protected $routeMatch;
+
+  /**
+   * Constructs a new PaymentListBuilder object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeInterface $entity_type
+   *   The entity type definition.
+   * @param \Drupal\Core\Entity\EntityStorageInterface $storage
+   *   The entity storage class.
+   * @param \Drupal\Core\Routing\RouteMatchInterface $route_match
+   *   The current route match.
+   */
+  public function __construct(EntityTypeInterface $entity_type, EntityStorageInterface $storage,  RouteMatchInterface $route_match) {
+    parent::__construct($entity_type, $storage);
+
+    $this->routeMatch = $route_match;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function createInstance(ContainerInterface $container, EntityTypeInterface $entity_type) {
+    return new static(
+      $entity_type,
+      $container->get('entity.manager')->getStorage($entity_type->id()),
+      $container->get('current_route_match')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function load() {
+    $order = $this->routeMatch->getParameter('commerce_order');
+    return $this->storage->loadMultipleByOrder($order);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getOperations(EntityInterface $entity) {
+    /** @var \Drupal\commerce_shipping\Entity\Shipment $entity */
+    $operations = parent::getOperations($entity);
+
+    if (isset($operations['edit'])) {
+      $operations['edit'] = [
+        'title' => $this->t('Edit'),
+        'weight' => 30,
+        'url' => $entity->toUrl('edit-form')->setRouteParameter('commerce_order', $entity->getOrderId()),
+      ];
+    }
+
+    return $operations;
+  }
```

Comment:
**OPTIONAL CHANGE REQUEST**

*Readability*

Similar readability comments as above. It's good to minimize indented code when possible; it is more readable and it will also make lines shorter reducing line-length coding standards violations.

```
$operations = parent::getOperations($entity);
if (!isset($operations['edit'])) {
  return $operations;
}

// Code here.
```

Also notice that this is not very readable, and it violates line-length coding standards:

```
$operations['edit'] = [
  'title' => $this->t('Edit'),
  'weight' => 30,
  'url' => $entity->toUrl('edit-form')->setRouteParameter('commerce_order', $entity->getOrderId()),
];
```

This one is better:

```
$url = $entity->toUrl('edit-form')->setRouteParameter(
  'commerce_order',
  $entity->getOrderId()
);
$operations['edit'] = [
  'title' => $this->t('Edit'),
  'weight' => 30,
  'url' => $url,
];
```